doc/nrf/config_and_build/board_support/board_names.rst

```diff
@@ -8,23 +8,39 @@ Board names
 
-The build target column uses several entries for multi-core hardware platforms:
+The board targets follow Zephyr's :ref:`board_terminology` scheme and are used mostly when :ref:`building`.
+For example, the board target ``nrf54l15pdk@0.3.0/nrf54l15/cpuapp`` can be read as made of the following elements:
 
-* For core type:
++-------------+----------------+-------------------------+---------------------------------+
+| nrf54l15pdk |     @0.3.0     |        /nrf54l15        |             /cpuapp             |
++=============+================+=========================+=================================+
+| Board name  | Board revision | Board qualifier for SoC | Board qualifier for CPU cluster |
++-------------+----------------+-------------------------+---------------------------------+
 
-  * ``cpuapp`` - When you choose this target, you build the application core firmware.
-  * ``cpunet`` - When you choose this target, you build the network core firmware.
+While the board name is always present, other elements, such as the board revision or the board qualifiers, are optional:
 
-* For usage of Cortex-M Security Extensions (CMSE):
+* :ref:`Board revision <zephyr:glossary>` - You can use the *board_name@board_revision* board target to get extra devicetree overlays with new features available for a specific board version.
+  The *board_revision* is printed on the label of your DK, just below the PCA number.
+  For example, :ref:`building <building>` for the ``nRF9160dk@1.0.0/nrf9160`` board target adds the external flash on the nRF9160 DK that was available since :ref:`board version v0.14.0 <nrf9160_board_revisions>`.
 
-  * Entries without ``*/ns`` (``cpuapp``) - When you choose this target, you build the application core firmware as a single execution environment that does not use CMSE (:ref:`Trusted Firmware-M (TF-M) <ug_tfm>`).
+* :term:`System on Chip (SoC)` - When you choose a board target with this qualifier, you build for this specific SoC on this board.
+
+* :ref:`CPU cluster <zephyr:glossary>` - You can use this board qualifier to build for a group of one or more CPU cores, all executing the same image within the same address space and in a symmetrical (SMP) configuration.
+  The CPU cluster board qualifier varies depending on the SoC and SoC Series.
+  For example, :ref:`building <building>` for the ``thingy53/nrf5340/cpunet`` board target builds the network core firmware for Nordic Thingy:53, building for ``nrf54h20dk/nrf54h20/cpuapp`` builds the application core firmware for the nRF54H20 DK, building for ``nrf54h20dk/nrf54h20/cpurad`` builds the radio core firmware for the nRF54H20 DK, and so on.
+  Check the Product Specification of the given SoC for more information about the available CPU clusters.
+
+* :ref:`Variant <zephyr:glossary>` - You can use this board qualifier to build for a particular type or configuration of a build for a combination of SoC and CPU cluster.
```

Comment:
Should variant be in the table?

Reply:
There's no viable example. But I've added an empty field.